akd/src/append_only_zks.rs

```diff
@@ -340,6 +340,7 @@ impl Azks {
     /// before it is written to storage. The is_new flag indicates whether the
     /// returned node is new or not.
     #[async_recursion]
+    #[allow(clippy::multiple_bound_locations)]
```

Comment:
What is this for?

Reply:
Hmm, looks like cases where a [generic is defined in a bound predicate and a `where` clause](https://rust-lang.github.io/rust-clippy/master/index.html#/multiple_bound_locations). That said, I don't see that in this signature.

@haochenuw is this from a macro or something?

Reply:
Yes, this is to address a clippy warning. It seems to show up only for the functions with #[async_recursion].